5-AccessControl/2-call-api-groups/AppCreationScripts/Configure.ps1

```diff
@@ -0,0 +1,422 @@
+   Add-Content -Value "<tr><td>client</td><td>$currentAppId</td><td><a href='$clientPortalUrl'>msal-react-spa</a></td></tr>" -Path createdApps.html
+
+   $requiredResourcesAccess = New-Object System.Collections.Generic.List[Microsoft.Open.AzureAD.Model.RequiredResourceAccess]
+
+   # Add Required Resources Access (from 'client' to 'service')
+   Write-Host "Getting access from 'client' to 'service'"
+   $requiredPermissions = GetRequiredPermissions -applicationDisplayName "msal-node-api" `
+                                                -requiredDelegatedPermissions "access_as_user" `
+
+   $requiredResourcesAccess.Add($requiredPermissions)
+
+   # Add Required Resources Access (from 'client' to 'Microsoft Graph')
+   Write-Host "Getting access from 'client' to 'Microsoft Graph'"
+   $requiredPermissions = GetRequiredPermissions -applicationDisplayName "Microsoft Graph" `
+                                                -requiredDelegatedPermissions "User.Read|GroupMember.Read.All" `
+
+   $requiredResourcesAccess.Add($requiredPermissions)
+
+
+   Set-AzureADApplication -ObjectId $clientAadApplication.ObjectId -RequiredResourceAccess $requiredResourcesAccess
+   Write-Host "Granted permissions."
+
+   # Configure known client applications for service 
+   Write-Host "Configure known client applications for the 'service'"
+   $knowApplications = New-Object System.Collections.Generic.List[System.String]
+    $knowApplications.Add($clientAadApplication.AppId)
+   Set-AzureADApplication -ObjectId $serviceAadApplication.ObjectId -KnownClientApplications $knowApplications
+   Write-Host "Configured."
+
+   if($null -eq (Get-AzureADGroup -SearchString "GroupAdmin")) {
+    Write-Host "Creating group"
+    $newsg = New-AzureADGroup -Description "GroupAdmin"  -DisplayName "GroupAdmin" -MailEnabled $false -SecurityEnabled $true -MailNickName "GroupAdmin"
+    Write-Host "Successfully created $($newsg.DisplayName)"
+   }
+
+   Write-Host $creds
+   if($null -eq (Get-AzureADGroup -SearchString "GroupMember")) {
+    Write-Host "Creating group"
+    $newsg = New-AzureADGroup -Description "GroupMember"  -DisplayName "GroupMember" -MailEnabled $false -SecurityEnabled $true -MailNickName "GroupMember"
+    Write-Host "Successfully created $($newsg.DisplayName)"
+   }
+
+   $groupAdmin = Get-AzureADGroup -SearchString "GroupAdmin"
+   $groupMember = Get-AzureADGroup -SearchString "GroupMember"
+
+   # Update config file for 'service'
+   $configFile = $pwd.Path + "\..\API\authConfig.json"
+   Write-Host "Updating the sample code ($configFile)"
+   $dictionary = @{ "Enter_the_Tenant_Info_Here" = $tenantId;"Enter_the_Application_Id_Here" = $serviceAadApplication.AppId;"Enter_the_Client_Secret_Here" = $serviceAppKey; "Enter_the_Object_Id_of_GroupAdmin_Group_Here" = $groupAdmin.objectId; "Enter_the_Object_Id_of_GroupMember_Group_Here" = $groupMember.objectId };
```

Comment:
>Enter_the_Object_Id_of_GroupAdmin_Group_Here" = $groupAdmin.objectId [](http://example.com/codeflow?start=174&length=68)

thank you

Reply:
Note: 

[4:16 p.m.] Kalyan Krishna
    the appcreation script populated one, but not the other
​[4:17 p.m.] Kalyan Krishna
    export const securityGroups = {​
GroupMember: "Enter_the_Object_Id_of_GroupMember_Here",
GroupAdmin: "1f32090b-3dd7-4728-b9fa-7a7bc60cbb61"
(1 liked)